src/tokenizer/tokenizer.py

```diff
@@ -648,6 +654,13 @@ def parse_digits(w):
         sec = int(p[2])
         if (0 <= h < 24) and (0 <= m < 60) and (0 <= sec < 60):
             return TOK.Time(w, h, m, sec), s.end()
+    s = re.match(r'\d+[A-Za-z]{1}', w)
+    if s:
```

Comment:
I suggest adding an `int(n)` in the NumberWithCharacter() constructor, i.e. storing the number as an integer, not as a string. Another possibility would simply be not to break down the string into two parts, but just to store it as-is. The choice depends on what is most useful for downstream processing of addresses.